packages/number-field/src/vaadin-number-field.js

```diff
@@ -0,0 +1,435 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { PolymerElement, html } from '@polymer/polymer';
+import { ElementMixin } from '@vaadin/vaadin-element-mixin/vaadin-element-mixin.js';
+import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+import { InputFieldMixin } from '@vaadin/field-base/src/input-field-mixin.js';
+import '@vaadin/input-container/src/vaadin-input-container.js';
+import '@vaadin/text-field/src/vaadin-input-field-shared-styles.js';
+
+export class NumberField extends InputFieldMixin(ThemableMixin(ElementMixin(PolymerElement))) {
+  static get is() {
+    return 'vaadin-number-field';
+  }
+
+  static get version() {
+    return '22.0.0-alpha1';
+  }
+
+  static get template() {
+    return html`
+      <style include="vaadin-input-field-shared-styles">
+        :host([readonly]) [part$='button'] {
+          pointer-events: none;
+        }
+
+        [part='decrease-button']::before {
+          content: '−';
+        }
+
+        [part='increase-button']::before {
+          content: '+';
+        }
+
+        [part='decrease-button'],
+        [part='increase-button'] {
+          -webkit-user-select: none;
+          -moz-user-select: none;
+          user-select: none;
+        }
+
+        :host([dir='rtl']) [part='input-field'] {
+          direction: ltr;
+        }
+
+        /* TODO: provide alternatives for Safari */
+        :host([dir='rtl']) [part='input-field'] ::slotted(input)::placeholder {
+          direction: rtl;
+        }
+
+        :host([dir='rtl']:not([has-controls])) [part='input-field'] ::slotted(input)::placeholder {
+          text-align: left;
+        }
+      </style>
+
+      <div part="container">
+        <div part="label" on-click="focus">
+          <slot name="label"></slot>
+        </div>
+
+        <vaadin-input-container
+          part="input-field"
+          readonly="[[readonly]]"
+          disabled="[[disabled]]"
+          invalid="[[invalid]]"
+          theme$="[[theme]]"
+        >
+          <div
+            disabled$="[[!_allowed(-1, value, min, max, step)]]"
+            part="decrease-button"
+            on-click="_decreaseValue"
+            on-touchend="_decreaseButtonTouchend"
+            hidden$="[[!hasControls]]"
+            slot="prefix"
+          ></div>
+          <slot name="prefix" slot="prefix"></slot>
+          <slot name="input"></slot>
+          <slot name="suffix" slot="suffix"></slot>
+          <div id="clearButton" part="clear-button" slot="suffix"></div>
+          <div
+            disabled$="[[!_allowed(1, value, min, max, step)]]"
```

Comment:
```suggestion
            disabled$="[[!_allowed(1)]]"
```

Reply:
Seems that the dependencies of the method still have to be specified here in order to have the disabled property recalculated after an argument has changed.

Reply:
This line is intended to trigger the observer when any of the arguments change.